docs/cryptodoc/src/05_10_cmce.rst

```diff
@@ -0,0 +1,469 @@
+   +--------------------------------------------------------+------------------------------------------------------------------+----------------------------------------------------------------------------+
+   | :ref:`Parameters <pubkey/cmce/params>`                 | :srcref:`[src/lib/pubkey/classic_mceliece]/cmce_parameters.h`    | Classic McEliece parameter container                                       |
+   +--------------------------------------------------------+------------------------------------------------------------------+----------------------------------------------------------------------------+
+   | Classic McEliece                                       | :srcref:`[src/lib/pubkey/classic_mceliece]/cmce.h`               | Classic McEliece public and private key interface                          |
+   +--------------------------------------------------------+------------------------------------------------------------------+----------------------------------------------------------------------------+
+
+
+.. _pubkey/cmce/types:
+
+Types
+^^^^^
+
+In implementing Botan's Classic McEliece, strong types are utilized to
+ensure the correct data usage within the code. These strong types are
+fundamental to separate the domain of the various byte sequences involved in
+Classic McEliece, such as multiple types of seeds, random byte sequences for different
+algorithms, and bit-vectors with different semantic contexts. Specific integers,
+like raw Galois Field elements and their modulus, are
+also represented as strong types. This reliance on strong types is similar to using them
+in :ref:`SPHINCS+ <signatures/sphincsplus/types>` for
+similar reasons. The Classic McEliece implementation enhances readability,
+clarity, and reliability by employing strong types.
+
+
+..  _pubkey/cmce/gf:
+
+Galois Field Arithmetic
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The Goppa code is based on the Galois field (GF) :math:`\mathbb{F}_{q}`, where :math:`q=2^m`.
+To represent GF elements, we utilize the class ``Classic_McEliece_GF``. Each
+element is defined by two polynomials: one defines its value, and the other
+defines the modulus of the field. These values are represented as unsigned
+integers. For example, the element :math:`z^3+z+1` is represented by the unsigned
+integer ``0b1011``.
+
+The GF elements support various operations such as addition, multiplication,
+and inversion. These operations are implemented in constant time for operands
+within the same field. Multiplication is performed using a constant-time
+schoolbook multiplication algorithm with a consecutive constant-time reduction.
+Inversion is achieved using Fermat's little theorem: :math:`a^{-1} = a^{q-2}`.
+For that, exponentiation is implemented using a simple square-and-multiply
+algorithm. It is worth noting that square-and-multiply has no timing
```

Comment:
Then I'd just say "a constant-time squate-and-multiply", which is fair because in this context the algorithm is constant time as the exponent is constant. (I just expect comments on whether the algorithm is implemented constant-time if we mention just the algorithm).